From 6d51a9e510247b41c5ec864c864ed197af7cd085 Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Thu, 26 Apr 2012 12:15:36 +0300
Subject: [PATCH 149/370] conf/obj.c: extend `Dependencies'

Add the "configuration database is not that big" assumption.

RB: r/765
---
 conf/obj.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/conf/obj.c b/conf/obj.c
index 2646685..7b8eeff 100644
--- a/conf/obj.c
+++ b/conf/obj.c
@@ -140,6 +140,11 @@
  * <hr> <!------------------------------------------------------------>
  * @section conf-depends Dependencies
  *
+ * - We assume that the size of configuration database is very small
+ *   compared with other meta-data.  Confd can take advantage of this
+ *   assumption and load the entire contents of configuration database
+ *   into the cache.
+ *
  * - c2_rpc_item_get() and c2_rpc_item_put() should be implemented.
  *
  *   Confc implementation schedules a state transition in
-- 
1.8.3.2

